src/CoreBundle/DataAccess/DataAccessCmsMasterPagedefDatabase.php

```diff
@@ -25,11 +25,12 @@ public function __construct(DataAccessCmsMasterPagedefInterface $fallbackLoader,
         $this->inputFilterUtil = $inputFilterUtil;
     }
 
-    public function get(string $id): ?CmsMasterPagdef
+    public function get(string $id, ?string $type = null): ?CmsMasterPagdef
```

Comment:
we are changing the interface here - why? And if we are, can we do away with the default?

Also-  it looks to me as if we are passing the value deep into the chain before finally using it.... In addition, we are only using it in `DataAccessCmsMasterPagedefFile` - that would make me very hesitant to move the parameter into the interface.

If the file requires a type (but no other pagedef needs this), then I would suggest to simply add the type as part of the pagedef ID.

TYPE:ID for example. That way we can keep the interface as it is - and have `DataAccessCmsMasterPagedefFile` extract the type from the ID-String passed.

Reply:
Finally it ends up in DataAccessCmsMasterPagedefFile::getPageDefinitionFilePath() where the old implementation was (only) reading it from the request.

At least in those request only cases there is probably no sensible default.

As this is an interface change it would be a BC break to require an additional parameter I guess.

Reply:
"Hiding" the type in the id sounds hackish/wrong?

Reply:
(changed pull request target to master)